src/api.h

```diff
@@ -39,25 +34,325 @@ AWS_CRT_API const char *aws_crt_error_debug_str(int);
+    aws_crt_credentials_provider_ecs_options *options);
+
+/* X509 credentials provider */
+typedef struct _aws_crt_credentials_provider_x509_options aws_crt_credentials_provider_x509_options;
+AWS_CRT_API aws_crt_credentials_provider_x509_options *aws_crt_credentials_provider_x509_options_new(void);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_release(aws_crt_credentials_provider_x509_options *options);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_get_thing_name(
+    aws_crt_credentials_provider_x509_options *options,
+    uint8_t **out_thing_name,
+    size_t *out_thing_name_length);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_set_thing_name(
+    aws_crt_credentials_provider_x509_options *options,
+    uint8_t *thing_name,
+    size_t thing_name_length);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_get_role_alias(
+    aws_crt_credentials_provider_x509_options *options,
+    uint8_t **out_role_alias,
+    size_t *out_role_alias_length);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_set_role_alias(
+    aws_crt_credentials_provider_x509_options *options,
+    uint8_t *role_alias,
+    size_t role_alias_length);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_get_endpoint(
+    aws_crt_credentials_provider_x509_options *options,
+    uint8_t **out_endpoint,
+    size_t *out_endpoint_length);
+AWS_CRT_API void aws_crt_credentials_provider_x509_options_set_endpoint(
+    aws_crt_credentials_provider_x509_options *options,
+    uint8_t *endpoint,
+    size_t endpoint_length);
+AWS_CRT_API aws_crt_credentials_provider *aws_crt_credentials_provider_x509_new(
+    aws_crt_credentials_provider_x509_options *options);
+
+/* STS Web Identity provider */
+typedef struct aws_credentials_provider_sts_web_identity_options aws_crt_credentials_provider_sts_web_identity_options;
+AWS_CRT_API aws_crt_credentials_provider_sts_web_identity_options *
+    aws_crt_credentials_provider_sts_web_identity_options_new(void);
+AWS_CRT_API void aws_crt_credentials_provider_sts_web_identity_options_release(
+    aws_crt_credentials_provider_sts_web_identity_options *options);
+AWS_CRT_API aws_crt_credentials_provider *aws_crt_credentials_provider_sts_web_identity_new(
+    aws_crt_credentials_provider_sts_web_identity_options *options);
+
+/* Generic credentials provider release */
+AWS_CRT_API void aws_crt_credentials_provider_release(aws_crt_credentials_provider *credentials_provider);
+
+/* aws_signing_config_aws */
+typedef struct _aws_crt_signing_config_aws aws_crt_signing_config_aws;
+AWS_CRT_API aws_crt_signing_config_aws *aws_crt_signing_config_aws_new(void);
+AWS_CRT_API void aws_crt_signing_config_aws_release(aws_crt_signing_config_aws *signing_config);
+
+AWS_CRT_API int aws_crt_signing_config_aws_get_algorithm(aws_crt_signing_config_aws *signing_config);
+AWS_CRT_API void aws_crt_signing_config_aws_set_algorithm(aws_crt_signing_config_aws *signing_config, int algorithm);
```

Comment:
we'll eventually get enums for stuff like `algorithm`, right?

Reply:
Yes that's coming

Reply:
Why can't we just use the enum type directly?